### ISSUE TITLE
Empty strings in dateRangeInput()'s max or min cause problems

### ISSUE THREAD

**Issue by nathancday** [rstudio/shiny#2402]
When either `max` or `min` are set to an empty string `""` in `dateRangeInput` it seems that the server portion of the app will not execute. There are no error messages and the app appears to load fine, even the `dataRangeInput` is active. 

```
library(shiny)

ui <- fluidPage(
    titlePanel("Shiny Issue about dateRangeInput()"),
    sidebarLayout(
        sidebarPanel(
            dateRangeInput("dr", "Empty strings break things",
                           # min = "",
                           max = ""
            )
        ),
        mainPanel(
           plotOutput("hist")
        )
    )
)

server <- function(input, output) {
    observeEvent(input$dr, {
        print(input$dr)
    })
    output$hist <- renderPlot({
        hist(faithful$waiting)
    })
}

shinyApp(ui = ui, server = server)
```
Restoring the empty string to `NULL` or a valid format fixes things.

I know the `""` input is not valid, but it would be nice to have a check in place that produces a meaningful error message when this situation arises. I thought I was losing my mind de-bugging this, because I had not indication from R that something was amiss and tracking a server/rendering problem to the ui was also unexpected.

**Comment by alandipert** [comment]
Thank you for bringing this to our attention. Looking at `dateRangeInput()`, I think it might make the most sense for the function to produce an error if either `min` or `max` are character vectors in any other format than `"%Y-%m-%d"`.

What do you think about that instead of checking just for `""`?

In the meantime I'll consult with the team to figure out if either enhancement is a change we'd consider.

**Comment by alandipert** [comment]
I consulted with @wch and we decided checking the date strings for validity and producing warnings if they are invalid would be a solid improvement.

The approach we came up with is this:

1. Coerce any arguments to either `dateInput` or `dateRangeInput` to `Date` objects using `as.Date`
1. If the coercion of any argument fails, swallow the error and produce a helpful warning.

If you're interested in contributing changes along these lines, I'd be happy to help. Otherwise it's something we'll get to.
